library/ssl_misc.h

```diff
@@ -1743,4 +1751,57 @@ static inline const void *mbedtls_ssl_get_groups( const mbedtls_ssl_context *ssl
     #endif
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/* Corresponding PSA algorithm for MBEDTLS_CIPHER_NULL.
+ * Same value is used fo PSA_ALG_CATEGORY_CIPHER, hence it is
+ * guaranteed to not be a valid PSA algorithm identifier.
+ */
+#define MBEDTLS_SSL_NULL_CIPHER 0x04000000
+
+/**
+ * \brief       Translate mbedtls cipher type/taglen pair to psa:
+ *              algorithm, key type and key size.
+ *
+ * \param  mbedtls_cipher_type [in] given mbedtls cipher type
+ * \param  taglen              [in] given tag length
+ *                                  0 - default tag length
+ * \param  alg                 [out] corresponding PSA alg
+ *                                   There is no corresponding PSA
+ *                                   alg for MBEDTLS_SSL_NULL_CIPHER, so
+ *                                   MBEDTLS_SSL_NULL_CIPHER is returned
+ * \param  key_type            [out] corresponding PSA key type
+ * \param  key_size            [out] corresponding PSA key size
+ *
+ * \return                     PSA_SUCCESS on success or PSA_ERROR_NOT_SUPPORTED if
+ *                             conversion is not supported.
+ */
+psa_status_t mbedtls_ssl_cipher_to_psa( mbedtls_cipher_type_t mbedtls_cipher_type,
+                                    size_t taglen,
+                                    psa_algorithm_t *alg,
+                                    psa_key_type_t *key_type,
+                                    size_t *key_size );
+
+/**
+ * \brief       Convert given PSA status to mbedtls error code.
+ *
+ * \param  status      [in] given PSA status
+ *
+ * \return             corresponding mbedtls error code
+ */
+static inline int psa_status_to_mbedtls( psa_status_t status )
```

Comment:
Just noticed: since this is an internal function of the SSL module, its name should be prefixed with `ssl_`. (Also, we should leave the `psa_` namespace to actual PSA function, so functions that are related to PSA but not part of it should start with `mbedtls_psa_` if they're visible of `modulename_psa` (in that case, `ssl_psa`) when they're static.)

Reply:
`ssl_psa_status_to_mbedtls()` does not meet required patter:
`(mbedtls|psa)_[0-9a-z_]*[0-9a-z]$`

Maybe: `psa_ssl_status_to_mbedtls()` ?